src/components/modals/Dialog.jsx

```diff
@@ -0,0 +1,120 @@
+import React, { useLayoutEffect } from 'react';
+import PropTypes from 'prop-types';
+import ReactDOM from 'react-dom';
+import { CSSTransition } from 'react-transition-group';
+import FocusTrap from 'focus-trap-react';
+import noScroll from 'no-scroll';
+import DenyIcon from '../../../lib/icons/react/DenyIcon';
+import Flex from '../layout/Flex';
+import FlexItem from '../layout/FlexItem';
+import Section from '../layout/Section';
+
+const Dialog = (props) => {
+  const handleClick = () => {
+    if (props.canDismiss) {
+      props.onDismiss();
+    }
+  };
+
+  const handleKeyDown = (e) => {
+    if (props.canDismiss && e.key === 'Escape') {
+      props.onDismiss();
+    }
+  };
+
+  useLayoutEffect(() => {
+    // This toggles scrolling on and off based on whether the modal
+    // is shown or not
+    if (props.isOpen) noScroll.on();
+    noScroll.off();
+  }, [props.isOpen]);
+
+  return ReactDOM.createPortal(
+    <CSSTransition timeout={200} in={props.isOpen} classNames="dialog" unmountOnExit>
+      <React.Fragment>
+        <FocusTrap>
+          <div>
+            <div className="dialog-overlay" onClick={handleClick}></div>
+            <div
+              className="dialog elevation--3"
+              role={props.role}
+              aria-labelledby={props.title && 'a11y-dialog-title'}
+              aria-describedby="a11y-dialog-desc"
+              tabIndex="-1"
+              aria-modal="true"
+              onKeyDown={handleKeyDown}
+            >
+              {(props.title || props.canDismiss) && (
+                <React.Fragment>
+                  <div className="dialog-header">
+                    <Section border="bottom">
+                      <Flex justify="spaceBetween" align="start">
+                        <FlexItem>
+                          {props.title && (
+                            <div
+                              className="dialog-title type--head4"
+                              id="a11y-dialog-title"
+                            >
+                              {props.title}
+                            </div>
+                          )}
+                        </FlexItem>
+                        {props.canDismiss && (
+                          <FlexItem shrink>
+                            <button className="dialog-icon" onClick={handleClick}>
+                              <DenyIcon size="xs" />
+                            </button>
+                          </FlexItem>
+                        )}
+                      </Flex>
+                    </Section>
+                  </div>
+                </React.Fragment>
+              )}
+              <div className="dialog-content">
+                <Section>{props.content}</Section>
+              </div>
+              {props.footerContent && (
```

Comment:
Looking at `Prompt`, if the bottom buttons are passed in `props.content` which then toggles `props.isOpen` outside of this component, then when will `props.footerContent` be used?

Reply:
Yeah that's a very good question, with a complicated answer:

footerContent is for modals that are larger, so that... when the modal scrolls, the buttons aren't hidden at the bottom (if that happens, we'd make developers construct that modal manually by hand). The reason buttons aren't in the footer now, is because the footer is gray.

That being said, we have an action item to talk with design regarding this. We may always want buttons in the footer, but we don't necessary always want them on a gray background, so we may require some sort of theme prop to control whether the footer will be gray or white, and then the buttons will appear the same as they do now, except they'll actually be in the footer.